### ISSUE TITLE
Produces broken code when formatting trait aliases

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rustfmt#3734]
```diff
-trait MySource<T> = Source<Item = T, Error: Debug> + Send + 'static;
-trait MySink<T> = Sink<Item = T, Error: Debug> + Send + 'static;
+trait MySource<T> = Source<Item = T, Error = Debug> + Send + 'static;
+trait MySink<T> = Sink<Item = T, Error = Debug> + Send + 'static;
```

**Comment by nagisa** [comment]
The version we are using is rustfmt 1.3.3-nightly (66c27c9 2019-07-16).

**Comment by topecongiro** [comment]
Dup of #3657. This is now fixed in rustfmt 1.4.4-nightly.
